src/dir.targets

```diff
@@ -160,8 +147,35 @@
 
   <!-- On command line run unit tests on CoreCLR after the Test target -->
   <Target Name="RunTestsForProject"
-          AfterTargets="Test"
-          Condition="'$(RunTestsForProject)'=='true'">
+          AfterTargets="CheckTestCategories"
+          Condition="'$(TestDisabled)'!='True'">
+
+    <PropertyGroup Condition="'$(RunInnerLoop)'=='True'">
+      <!-- When inner-loop tests are enabled we want to exclude categories -->
+      <XunitOptions Condition="'$(RunOuterLoop)'!='True'">$(XunitOptions) -notrait category=outerloop</XunitOptions>
+    </PropertyGroup>
+    <PropertyGroup Condition="'$(RunInnerLoop)'!='True'">
+      <!-- When inner-loop tests are disabled we want to include categories -->
+      <XunitOptions Condition="'$(RunOuterLoop)'=='True'">$(XunitOptions) -trait category=outerloop</XunitOptions>
+    </PropertyGroup>
```

Comment:
Should just do something like:
```
 <XunitOptions>$(XunitOptions) @(TestCategories -> '-trait category=%(Identity)')</XunitOptions>
```

That would probably allow you to remove most of the Is/RunInnerLoop properties. And you can filter the TestTraits/Categories via the RunTestTraits item.